Demo / Example does not compile

First I tried your processor tests, everything ok.
Then I tried the first example from your article 1, generated artifact, ok.
Then I tried to compile this repository, results in error:

Directory: ../work-with-j2cl-app-from-js/
mvn clean,   ok
mvn j2cl:watch, results in following error log:

[INFO] Scanning for projects...
[INFO]
[INFO] --------< org.treblereel.j2cl.helloworld:simple-xml-marshaller >--------
[INFO] Building simple-xml-marshaller 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- j2cl-maven-plugin:0.19:watch (default-cli) @ simple-xml-marshaller ---
[INFO] Using /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/simple-xml-marshaller-1.0-SNAPSHOT as webappDirectory since goal is running in a non-reactor build and none was set
[INFO] Removing dependency com.google.jsinterop:base:jar:1.0.0:compile, replacing with com.vertispan.jsinterop:base:jar:1.0.0-1
[INFO] Removing dependency com.google.jsinterop:base:jar:1.0.0:compile, replacing with com.vertispan.jsinterop:base:jar:1.0.0-1
[INFO] Removing dependency com.google.jsinterop:base:jar:1.0.0:compile, replacing with com.vertispan.jsinterop:base:jar:1.0.0-1
[INFO] Removing dependency com.google.jsinterop:base:jar:1.0.0:compile, replacing with com.vertispan.jsinterop:base:jar:1.0.0-1
[INFO] Removing dependency com.google.jsinterop:base:jar:1.0.0:compile, replacing with com.vertispan.jsinterop:base:jar:1.0.0-1
[INFO] Removing dependency com.google.jsinterop:base:jar:1.0.0:compile, replacing with com.vertispan.jsinterop:base:jar:1.0.0-1
[INFO] Starting org.treblereel.gwt.xml.mapper:api:0.6/unpack
[INFO] Finished org.treblereel.gwt.xml.mapper:api:0.6/unpack in 25ms
[INFO] Starting org.treblereel.gwt.xml.mapper:common:0.6/unpack
[INFO] Finished org.treblereel.gwt.xml.mapper:common:0.6/unpack in 17ms
[INFO] Starting com.google.elemental2:elemental2-dom:1.1.0/unpack
[INFO] Finished com.google.elemental2:elemental2-dom:1.1.0/unpack in 71ms
[INFO] Starting com.google.jsinterop:jsinterop-annotations:2.0.0/unpack
[INFO] Starting com.vertispan.jsinterop:base:1.0.0-1/unpack
[INFO] Starting com.google.elemental2:elemental2-core:1.1.0/unpack
[INFO] Finished com.google.elemental2:elemental2-core:1.1.0/unpack in 13ms
[INFO] Starting com.google.elemental2:elemental2-promise:1.1.0/unpack
[INFO] Starting org.treblereel.gwt.xml.mapper:processor:0.6/unpack
[INFO] Finished org.treblereel.gwt.xml.mapper:processor:0.6/unpack in 113ms
[INFO] Starting org.treblereel.j2cl.processors:processors:0.5/unpack
[INFO] Finished org.treblereel.j2cl.processors:processors:0.5/unpack in 111ms
[INFO] Starting org.treblereel.j2cl.processors:annotations:0.5/unpack
[INFO] Starting com.fasterxml.woodstox:woodstox-core:6.2.1/unpack
[INFO] Finished com.fasterxml.woodstox:woodstox-core:6.2.1/unpack in 62ms
[INFO] Starting org.codehaus.woodstox:stax2-api:4.2.1/unpack
[INFO] Finished org.codehaus.woodstox:stax2-api:4.2.1/unpack in 8ms
[INFO] Starting org.apache.felix:org.osgi.core:1.4.0/unpack
[INFO] Start watching {org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT=[/home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/src/main/java, /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/src/main/resources]}
[INFO] Watching for changes
[INFO] Starting com.fasterxml.woodstox:woodstox-core:6.2.1/bytecode
[INFO] Starting com.vertispan.jsinterop:base:1.0.0-1/bytecode
[INFO] Starting com.google.elemental2:elemental2-dom:1.1.0/bytecode
[INFO] Starting com.vertispan.jsinterop:base:1.0.0-1/stripped_sources
[INFO] Finished com.fasterxml.woodstox:woodstox-core:6.2.1/bytecode in 61ms
[INFO] Finished com.google.elemental2:elemental2-dom:1.1.0/bytecode in 70ms
[INFO] Finished com.vertispan.jsinterop:base:1.0.0-1/stripped_sources in 89ms
[INFO] Starting org.treblereel.gwt.xml.mapper:processor:0.6/bytecode
[INFO] Starting com.google.jsinterop:jsinterop-annotations:2.0.0/bytecode
[INFO] Starting org.treblereel.gwt.xml.mapper:api:0.6/bytecode
[INFO] Starting org.treblereel.j2cl.processors:processors:0.5/bytecode
[INFO] Finished org.treblereel.gwt.xml.mapper:api:0.6/bytecode in 12ms
[INFO] Starting org.treblereel.gwt.xml.mapper:common:0.6/bytecode
[INFO] Finished org.treblereel.gwt.xml.mapper:common:0.6/bytecode in 13ms
[INFO] Starting org.apache.felix:org.osgi.core:1.4.0/bytecode
[INFO] Finished org.apache.felix:org.osgi.core:1.4.0/bytecode in 6ms
[INFO] Starting com.google.elemental2:elemental2-promise:1.1.0/bytecode
[INFO] Finished org.treblereel.gwt.xml.mapper:processor:0.6/bytecode in 100ms
[INFO] Starting org.treblereel.j2cl.processors:annotations:0.5/bytecode
[INFO] Starting com.google.elemental2:elemental2-core:1.1.0/bytecode
[INFO] Finished com.google.elemental2:elemental2-core:1.1.0/bytecode in 9ms
[INFO] Starting org.codehaus.woodstox:stax2-api:4.2.1/bytecode
[INFO] Finished org.treblereel.j2cl.processors:processors:0.5/bytecode in 100ms
[INFO] Finished org.codehaus.woodstox:stax2-api:4.2.1/bytecode in 8ms
[INFO] Starting com.fasterxml.woodstox:woodstox-core:6.2.1/stripped_sources
[INFO] Starting com.vertispan.jsinterop:base:1.0.0-1/transpiled_js
[INFO] Starting com.vertispan.jsinterop:base:1.0.0-1/stripped_bytecode
[INFO] Starting com.google.elemental2:elemental2-dom:1.1.0/stripped_sources
[INFO] Finished com.google.elemental2:elemental2-dom:1.1.0/stripped_sources in 40ms
[INFO] Starting com.google.jsinterop:jsinterop-annotations:2.0.0/stripped_sources
[INFO] Starting org.treblereel.gwt.xml.mapper:api:0.6/stripped_sources
[INFO] Finished org.treblereel.gwt.xml.mapper:api:0.6/stripped_sources in 11ms
[INFO] Starting org.treblereel.gwt.xml.mapper:common:0.6/stripped_sources
[INFO] Finished org.treblereel.gwt.xml.mapper:common:0.6/stripped_sources in 261ms
[INFO] Starting org.apache.felix:org.osgi.core:1.4.0/stripped_sources
[INFO] Starting com.google.elemental2:elemental2-promise:1.1.0/stripped_sources
[INFO] Starting org.treblereel.j2cl.processors:annotations:0.5/stripped_sources
[INFO] Starting com.google.elemental2:elemental2-core:1.1.0/stripped_sources
[INFO] Starting org.codehaus.woodstox:stax2-api:4.2.1/stripped_sources
[INFO] Starting org.treblereel.gwt.xml.mapper:processor:0.6/stripped_sources
[INFO] Starting org.treblereel.j2cl.processors:processors:0.5/stripped_sources
[INFO] Starting org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/bytecode
[INFO] Finished com.vertispan.jsinterop:base:1.0.0-1/stripped_bytecode in 569ms
[INFO] Starting com.google.jsinterop:jsinterop-annotations:2.0.0/stripped_bytecode
[INFO] Starting com.google.jsinterop:jsinterop-annotations:2.0.0/transpiled_js
[INFO] Starting org.apache.felix:org.osgi.core:1.4.0/stripped_bytecode
[INFO] Starting org.treblereel.j2cl.processors:annotations:0.5/transpiled_js
Deserializers generation started
Serializers generation started
[INFO] Finished org.treblereel.j2cl.processors:annotations:0.5/transpiled_js in 233ms
[INFO] Starting org.treblereel.j2cl.processors:annotations:0.5/stripped_bytecode
   Generating UserBeanXMLSerializerImpl
   Writing UserBeanXMLSerializerImpl
   Generating UserBeanXMLDeserializerImpl
[INFO] Finished org.treblereel.j2cl.processors:annotations:0.5/stripped_bytecode in 42ms
[INFO] Starting org.codehaus.woodstox:stax2-api:4.2.1/stripped_bytecode
   Writing UserBeanXMLDeserializerImpl
Writing User_XMLMapperImpl
[INFO] Starting org.treblereel.gwt.xml.mapper:processor:0.6/stripped_bytecode
[INFO] Finished com.vertispan.jsinterop:base:1.0.0-1/transpiled_js in 919ms
[INFO] Starting org.treblereel.j2cl.processors:processors:0.5/transpiled_js
[INFO] Starting org.treblereel.j2cl.processors:processors:0.5/stripped_bytecode
[INFO] Starting com.vertispan.jsinterop:base:1.0.0-1/stripped_bytecode_headers
[INFO] Starting com.google.jsinterop:jsinterop-annotations:2.0.0/stripped_bytecode_headers
[INFO] Starting org.apache.felix:org.osgi.core:1.4.0/stripped_bytecode_headers
[INFO] Starting org.treblereel.j2cl.processors:annotations:0.5/stripped_bytecode_headers
[INFO] Starting org.codehaus.woodstox:stax2-api:4.2.1/stripped_bytecode_headers
[INFO] Starting org.treblereel.gwt.xml.mapper:processor:0.6/stripped_bytecode_headers
[INFO] Starting org.treblereel.j2cl.processors:processors:0.5/stripped_bytecode_headers
[INFO] Starting com.google.elemental2:elemental2-promise:1.1.0/transpiled_js
[INFO] Starting com.google.elemental2:elemental2-promise:1.1.0/stripped_bytecode
[INFO] com.google.elemental2:elemental2-promise:1.1.0/stripped_bytecode: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/com.google.elemental2-elemental2-promise-1.1.0/589a72085f7d1d8e4b40b11068a48f46-stripped_sources/results/elemental2/promise/Promise.java /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/com.google.elemental2-elemental2-promise-1.1.0/589a72085f7d1d8e4b40b11068a48f46-stripped_sources/results/elemental2/promise/Promise.java uses unchecked or unsafe operations.
[INFO] com.google.elemental2:elemental2-promise:1.1.0/stripped_bytecode: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/com.google.elemental2-elemental2-promise-1.1.0/589a72085f7d1d8e4b40b11068a48f46-stripped_sources/results/elemental2/promise/Promise.java Recompile with -Xlint:unchecked for details.
[INFO] Finished com.google.elemental2:elemental2-promise:1.1.0/stripped_bytecode in 78ms
[INFO] Starting com.fasterxml.woodstox:woodstox-core:6.2.1/stripped_bytecode
[INFO] Starting com.google.elemental2:elemental2-promise:1.1.0/stripped_bytecode_headers
[INFO] Starting com.fasterxml.woodstox:woodstox-core:6.2.1/stripped_bytecode_headers
[INFO] Starting com.google.elemental2:elemental2-core:1.1.0/stripped_bytecode
[INFO] Finished com.google.elemental2:elemental2-promise:1.1.0/transpiled_js in 174ms
[INFO] Starting com.google.elemental2:elemental2-core:1.1.0/transpiled_js
[INFO] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/bytecode: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/src/main/java/org/treblereel/j2cl/helloworld/App.java /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/src/main/java/org/treblereel/j2cl/helloworld/App.java uses unchecked or unsafe operations.
[INFO] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/bytecode: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/src/main/java/org/treblereel/j2cl/helloworld/App.java Recompile with -Xlint:unchecked for details.
[INFO] Finished org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/bytecode in 693ms
[INFO] Starting org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/stripped_sources
[INFO] com.google.elemental2:elemental2-core:1.1.0/stripped_bytecode: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/com.google.elemental2-elemental2-core-1.1.0/7fda6f305cec8e11c845d98992d44726-stripped_sources/results/elemental2/core/JsArray.java /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/com.google.elemental2-elemental2-core-1.1.0/7fda6f305cec8e11c845d98992d44726-stripped_sources/results/elemental2/core/JsArray.java uses unchecked or unsafe operations.
[INFO] com.google.elemental2:elemental2-core:1.1.0/stripped_bytecode: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/com.google.elemental2-elemental2-core-1.1.0/7fda6f305cec8e11c845d98992d44726-stripped_sources/results/elemental2/core/JsArray.java Recompile with -Xlint:unchecked for details.
[INFO] Finished com.google.elemental2:elemental2-core:1.1.0/stripped_bytecode in 529ms
[INFO] Starting com.google.elemental2:elemental2-core:1.1.0/stripped_bytecode_headers
[INFO] Finished com.google.elemental2:elemental2-core:1.1.0/stripped_bytecode_headers in 8ms
[INFO] Starting com.google.elemental2:elemental2-dom:1.1.0/transpiled_js
[INFO] Starting com.google.elemental2:elemental2-dom:1.1.0/stripped_bytecode
[INFO] Finished com.google.elemental2:elemental2-core:1.1.0/transpiled_js in 1148ms
[INFO] com.google.elemental2:elemental2-dom:1.1.0/stripped_bytecode: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/com.google.elemental2-elemental2-dom-1.1.0/b26b4dcc0e2dbd8f83778e1c604f8ac1-stripped_sources/results/elemental2/dom/DOMRectReadOnly.java Some input files use or override a deprecated API.
[INFO] com.google.elemental2:elemental2-dom:1.1.0/stripped_bytecode: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/com.google.elemental2-elemental2-dom-1.1.0/b26b4dcc0e2dbd8f83778e1c604f8ac1-stripped_sources/results/elemental2/dom/DOMRectReadOnly.java Recompile with -Xlint:deprecation for details.
[INFO] Finished com.google.elemental2:elemental2-dom:1.1.0/stripped_bytecode in 916ms
[INFO] Starting com.google.elemental2:elemental2-dom:1.1.0/stripped_bytecode_headers
[INFO] Finished com.google.elemental2:elemental2-dom:1.1.0/stripped_bytecode_headers in 58ms
[INFO] Starting org.treblereel.gwt.xml.mapper:common:0.6/stripped_bytecode
[INFO] Starting org.treblereel.gwt.xml.mapper:common:0.6/transpiled_js
[INFO] org.treblereel.gwt.xml.mapper:common:0.6/stripped_bytecode: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.gwt.xml.mapper-common-0.6/cfe201679ea45f8d28c56e8349023cc4-stripped_sources/results/org/treblereel/gwt/xml/mapper/api/AbstractObjectMapper.java Some input files use unchecked or unsafe operations.
[INFO] org.treblereel.gwt.xml.mapper:common:0.6/stripped_bytecode: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.gwt.xml.mapper-common-0.6/cfe201679ea45f8d28c56e8349023cc4-stripped_sources/results/org/treblereel/gwt/xml/mapper/api/AbstractObjectMapper.java Recompile with -Xlint:unchecked for details.
[INFO] Finished org.treblereel.gwt.xml.mapper:common:0.6/stripped_bytecode in 302ms
[INFO] Starting org.treblereel.gwt.xml.mapper:common:0.6/stripped_bytecode_headers
[INFO] Finished org.treblereel.gwt.xml.mapper:common:0.6/stripped_bytecode_headers in 8ms
[INFO] Starting org.treblereel.gwt.xml.mapper:api:0.6/transpiled_js
[INFO] Finished org.treblereel.gwt.xml.mapper:common:0.6/transpiled_js in 852ms
[INFO] Starting org.treblereel.gwt.xml.mapper:api:0.6/stripped_bytecode
[INFO] Finished com.google.elemental2:elemental2-dom:1.1.0/transpiled_js in 2028ms
[INFO] org.treblereel.gwt.xml.mapper:api:0.6/stripped_bytecode: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.gwt.xml.mapper-api-0.6/91811045492ce22cc00885fc6b3f6e25-stripped_sources/results/org/treblereel/gwt/xml/mapper/api/deser/EnumXMLDeserializer.java Some input files use unchecked or unsafe operations.
[INFO] org.treblereel.gwt.xml.mapper:api:0.6/stripped_bytecode: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.gwt.xml.mapper-api-0.6/91811045492ce22cc00885fc6b3f6e25-stripped_sources/results/org/treblereel/gwt/xml/mapper/api/deser/EnumXMLDeserializer.java Recompile with -Xlint:unchecked for details.
[INFO] Finished org.treblereel.gwt.xml.mapper:api:0.6/stripped_bytecode in 384ms
[INFO] Starting org.treblereel.gwt.xml.mapper:api:0.6/stripped_bytecode_headers
[INFO] Finished org.treblereel.gwt.xml.mapper:api:0.6/stripped_bytecode_headers in 9ms
[INFO] Starting org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/transpiled_js
[INFO] Finished org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/transpiled_js in 109ms
[INFO] Finished org.treblereel.gwt.xml.mapper:api:0.6/transpiled_js in 1077ms
[INFO] Starting org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js
[WARNING] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/base.js:24:4: WARNING - [JSC_DUP_VAR_DECLARATION_TYPE_MISMATCH] variable $jscomp redefined with type {}, original definition at  [synthetic:input] :24 with type {
  ASSUME_ES5: boolean,
  ASSUME_NO_NATIVE_MAP: boolean,
  ASSUME_NO_NATIVE_SET: boolean,
  FORCE_POLYFILL_PROMISE: boolean,
  FORCE_POLYFILL_PROMISE_WHEN_NO_UNHANDLED_REJECTION: boolean,
  ISOLATE_POLYFILLS: boolean,
  IS_SYMBOL_NATIVE: boolean,
  POLYFILL_PREFIX: string,
  SIMPLE_FROUND_POLYFILL: boolean,
  TRUST_ES6_POLYFILLS: boolean,
  ...
}
var $jscomp = $jscomp || {};
    ^^^^^^^^^^^^^^^^^^^^^^^

[WARNING] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/base.js:27:0: WARNING - [JSC_TYPE_MISMATCH] assignment to property scope of $jscomp
found   : {}
required: {}
$jscomp.scope = {};
^^^^^^^^^^^^^^^^^^

[WARNING] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/base.js:27:8: WARNING - [JSC_CONSTANT_PROPERTY_REASSIGNED_VALUE] constant property scope assigned a value more than once
Initialized at  [synthetic:base] :27:8
$jscomp.scope = {};
        ^^^^^

[WARNING] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/base.js:27:8: WARNING - [JSC_DUP_VAR_DECLARATION_TYPE_MISMATCH] variable $jscomp.scope redefined with type {}, original definition at  [synthetic:input] :27 with type {}
$jscomp.scope = {};
        ^^^^^

..... a lot of warnings ...


[ERROR] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/util/defines.js:25:0: ERROR - [JSC_INVALID_DEFINE_LOCATION] @define must be initalized on a static qualified name in global or module scope
$jscomp.ASSUME_ES5 = false;
^^^^^^^^^^^^^^^^^^^^^^^^^^

[ERROR] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/util/defines.js:25:8: ERROR - [JSC_NON_CONST_DEFINE] @define $jscomp.ASSUME_ES5 has already been set at  [synthetic:util/defines] :25:8.
$jscomp.ASSUME_ES5 = false;
        ^^^^^^^^^^

[ERROR] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/util/defines.js:31:0: ERROR - [JSC_INVALID_DEFINE_LOCATION] @define must be initalized on a static qualified name in global or module scope
$jscomp.ASSUME_NO_NATIVE_MAP = false;
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[ERROR] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/util/defines.js:31:8: ERROR - [JSC_NON_CONST_DEFINE] @define $jscomp.ASSUME_NO_NATIVE_MAP has already been set at  [synthetic:util/defines] :31:8.
$jscomp.ASSUME_NO_NATIVE_MAP = false;
        ^^^^^^^^^^^^^^^^^^^^

[ERROR] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/util/defines.js:37:0: ERROR - [JSC_INVALID_DEFINE_LOCATION] @define must be initalized on a static qualified name in global or module scope
$jscomp.ASSUME_NO_NATIVE_SET = false;
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[ERROR] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/util/defines.js:37:8: ERROR - [JSC_NON_CONST_DEFINE] @define $jscomp.ASSUME_NO_NATIVE_SET has already been set at  [synthetic:util/defines] :37:8.
$jscomp.ASSUME_NO_NATIVE_SET = false;
        ^^^^^^^^^^^^^^^^^^^^

[ERROR] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/util/defines.js:47:0: ERROR - [JSC_INVALID_DEFINE_LOCATION] @define must be initalized on a static qualified name in global or module scope
$jscomp.SIMPLE_FROUND_POLYFILL = false;
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[ERROR] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/util/defines.js:47:8: ERROR - [JSC_NON_CONST_DEFINE] @define $jscomp.SIMPLE_FROUND_POLYFILL has already been set at  [synthetic:util/defines] :47:8.
$jscomp.SIMPLE_FROUND_POLYFILL = false;
        ^^^^^^^^^^^^^^^^^^^^^^

[ERROR] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/util/defines.js:56:0: ERROR - [JSC_INVALID_DEFINE_LOCATION] @define must be initalized on a static qualified name in global or module scope
$jscomp.ISOLATE_POLYFILLS = false;
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[ERROR] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/util/defines.js:56:8: ERROR - [JSC_NON_CONST_DEFINE] @define $jscomp.ISOLATE_POLYFILLS has already been set at  [synthetic:util/defines] :56:8.
$jscomp.ISOLATE_POLYFILLS = false;
        ^^^^^^^^^^^^^^^^^

[ERROR] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/util/defines.js:63:0: ERROR - [JSC_INVALID_DEFINE_LOCATION] @define must be initalized on a static qualified name in global or module scope
$jscomp.FORCE_POLYFILL_PROMISE = false;
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[ERROR] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/util/defines.js:63:8: ERROR - [JSC_NON_CONST_DEFINE] @define $jscomp.FORCE_POLYFILL_PROMISE has already been set at  [synthetic:util/defines] :63:8.
$jscomp.FORCE_POLYFILL_PROMISE = false;
        ^^^^^^^^^^^^^^^^^^^^^^

[ERROR] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/util/defines.js:70:0: ERROR - [JSC_INVALID_DEFINE_LOCATION] @define must be initalized on a static qualified name in global or module scope
$jscomp.FORCE_POLYFILL_PROMISE_WHEN_NO_UNHANDLED_REJECTION = false;
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[ERROR] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: /home/juergen/j2cl/Beginning_J2CL/work-with-j2cl-app-from-js/simple-xml-marshaller/target/gwt3BuildCache/0.19/org.treblereel.j2cl.processors-processors-0.5/4ad2aa83cc45029349182f8b640edc68-bytecode/results/com/google/javascript/jscomp/js/util/defines.js:70:8: ERROR - [JSC_NON_CONST_DEFINE] @define $jscomp.FORCE_POLYFILL_PROMISE_WHEN_NO_UNHANDLED_REJECTION has already been set at  [synthetic:util/defines] :70:8.
$jscomp.FORCE_POLYFILL_PROMISE_WHEN_NO_UNHANDLED_REJECTION = false;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[INFO] org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js: 14 error(s), 229 warning(s), 95.4% typed

[ERROR] Exception executing task org.treblereel.j2cl.helloworld:simple-xml-marshaller:1.0-SNAPSHOT/optimized_js
java.lang.IllegalStateException: Closure Compiler failed, check log for details
    at com.vertispan.j2cl.build.provided.ClosureTask$4.execute (ClosureTask.java:245)
    at com.vertispan.j2cl.build.TaskScheduler$2.executeTask (TaskScheduler.java:172)
    at com.vertispan.j2cl.build.TaskScheduler$2.lambda$onReady$0 (TaskScheduler.java:211)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:515)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run (ScheduledThreadPoolExecutor.java:304)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1128)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:628)
    at java.lang.Thread.run (Thread.java:829)
java.lang.RuntimeException
        at com.vertispan.j2cl.build.DiskCache.markFailed(DiskCache.java:452)
        at com.vertispan.j2cl.build.DiskCache$CacheResult.markFailure(DiskCache.java:54)
        at com.vertispan.j2cl.build.TaskScheduler$2.executeTask(TaskScheduler.java:181)
        at com.vertispan.j2cl.build.TaskScheduler$2.lambda$onReady$0(TaskScheduler.java:211)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)

```

--- a/helloworld/pom.xml
+++ b/helloworld/pom.xml
@@ -9,7 +9,7 @@
   <version>1.0-SNAPSHOT</version>
 
   <properties>
-    <j2cl.maven.plugin.version>0.19</j2cl.maven.plugin.version>
+    <j2cl.maven.plugin.version>0.20</j2cl.maven.plugin.version>
     <elemental2.version>1.1.0</elemental2.version>
 
     <jsinterop.base.version>1.0.0</jsinterop.base.version>

--- a/work-with-j2cl-app-from-js/bind-to-window-xml-marshaller/pom.xml
+++ b/work-with-j2cl-app-from-js/bind-to-window-xml-marshaller/pom.xml
@@ -12,7 +12,7 @@
     <artifactId>bind-to-window-xml-marshaller</artifactId>
 
     <properties>
-        <j2cl.maven.plugin.version>0.19</j2cl.maven.plugin.version>
+        <j2cl.maven.plugin.version>0.20</j2cl.maven.plugin.version>
         <elemental2.version>1.1.0</elemental2.version>
 
         <jsinterop.base.version>1.0.0</jsinterop.base.version>

--- a/work-with-j2cl-app-from-js/js-native-xml-marshaller/pom.xml
+++ b/work-with-j2cl-app-from-js/js-native-xml-marshaller/pom.xml
@@ -12,7 +12,7 @@
     <artifactId>js-native-xml-marshaller</artifactId>
 
     <properties>
-        <j2cl.maven.plugin.version>0.19</j2cl.maven.plugin.version>
+        <j2cl.maven.plugin.version>0.20</j2cl.maven.plugin.version>
         <elemental2.version>1.1.0</elemental2.version>
 
         <jsinterop.base.version>1.0.0</jsinterop.base.version>

--- a/work-with-j2cl-app-from-js/simple-xml-marshaller/pom.xml
+++ b/work-with-j2cl-app-from-js/simple-xml-marshaller/pom.xml
@@ -12,7 +12,7 @@
     <artifactId>simple-xml-marshaller</artifactId>
 
     <properties>
-        <j2cl.maven.plugin.version>0.19</j2cl.maven.plugin.version>
+        <j2cl.maven.plugin.version>0.20</j2cl.maven.plugin.version>
         <elemental2.version>1.1.0</elemental2.version>
 
         <jsinterop.base.version>1.0.0</jsinterop.base.version>
@@ -42,13 +42,13 @@
         <dependency>
             <groupId>org.treblereel.j2cl.processors</groupId>
             <artifactId>annotations</artifactId>
-            <version>0.4-SNAPSHOT</version>
+            <version>0.5</version>
         </dependency>
         <dependency>
             <groupId>org.treblereel.j2cl.processors</groupId>
             <artifactId>processors</artifactId>
             <scope>provided</scope>
-            <version>0.4-SNAPSHOT</version>
+            <version>0.5</version>
         </dependency>
     </dependencies>
 

